binrz/rz-test/rz-test.c

```diff
@@ -409,6 +420,23 @@ int rz_test_main(int argc, const char **argv) {
 		}
 	}
 
+	// filter out except_dir
+	if (rz_pvector_len(except_dir)) {
+		void **it;
+		rz_pvector_foreach (except_dir, it) {
+			char *p = rz_file_abspath_rel(cwd, (char *)*it);
+			for (ut32 i = 0; i < rz_pvector_len(&state.db->tests);) {
+				RzTest *test = rz_pvector_at(&state.db->tests, i);
+				if (rz_str_startswith(test->path, p)) {
+					rz_test_test_free(test);
+					rz_pvector_remove_at(&state.db->tests, i);
+					continue;
+				}
+				i++;
```

Comment:
why is this here and not in the `for`?

Reply:
Because no need to increment `i` if remove happens. It also works if put this in for, then performs `i--` when remove happens.